sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/ShardReadersPool.java

```diff
@@ -137,15 +189,16 @@ void stop() {
 
   boolean allShardsUpToDate() {
     boolean shardsUpToDate = true;
-    for (ShardRecordsIterator shardRecordsIterator : shardIteratorsMap.values()) {
+    for (ShardRecordsIterator shardRecordsIterator : shardIteratorsMap.get().values()) {
```

Comment:
Is it fine if shardIteratorsMap is changed to something else while this happens? I understand that this will effectively return the result for a previous snapshot of shardIteratorsMap, but - can any bad races happen as a result of that?

Reply:
I extracted shardIteratorsMap before the loop so there's no doubt that we want to use a consistent snapshot